File: core/src/main/java/pcb/uwu/unit/derived/fundamental/FrequencyUnit.java
package pcb.uwu.unit.derived.fundamental;

import pcb.uwu.core.CompositeUnit;
import pcb.uwu.core.UnitCounter;
import pcb.uwu.unit.base.TimeUnit;

public class FrequencyUnit extends CompositeUnit {

	public FrequencyUnit(TimeUnit time) {
		super(new UnitCounter()
				.minor(time));
	}
}

File: core/src/main/java/pcb/uwu/unit/derived/fundamental/FrequencyUnit.kt
package pcb.uwu.unit.derived.fundamental

import pcb.uwu.core.CompositeUnit
import pcb.uwu.core.UnitCounter
import pcb.uwu.unit.base.TimeUnit

open class FrequencyUnit : CompositeUnit
{
    @Suppress("ConvertSecondaryConstructorToPrimary")
    constructor(time: TimeUnit)
            : super(UnitCounter().minor(time))
}